src/main/java/com/epam/reportportal/junit/ParallelRunningHandler.java

```diff
@@ -377,6 +383,49 @@ protected FinishTestItemRQ buildFinishStepRq(FrameworkMethod method, String stat
 		return parameters.isEmpty() ? null : parameters;
 	}
 
+	protected Optional<Integer> getTestCaseId(FrameworkMethod method, Object runner, String codeRef) {
+		if (!(method.isStatic() || isIgnored(method))) {
+			Object target = LifecycleHooks.getTargetForRunner(runner);
+			return ofNullable(method.getMethod().getDeclaredAnnotation(TestCaseId.class)).flatMap(annotation -> {
+				if (annotation.isParameterized() && target instanceof ArtifactParams) {
+					return getParameterizedTestCaseId(target, codeRef);
+				} else {
+					return Optional.of(annotation.value());
+				}
+			});
+
+		}
+		return Optional.of(Objects.hashCode(codeRef));
+	}
+
+	protected Optional<Integer> getParameterizedTestCaseId(Object target, String codeRef) {
+
+		com.google.common.base.Optional<Map<String, Object>> params = ((ArtifactParams) target).getParameters();
+		if (params.isPresent()) {
+			return Arrays.stream(target.getClass().getDeclaredFields())
+					.filter(field -> params.get().containsKey(field.getName()) && field.getDeclaredAnnotation(TestCaseIdKey.class) != null)
+					.findFirst()
+					.map(testCaseIdField -> {
+						TestCaseIdKey testCaseIdKeyAnnotation = testCaseIdField.getDeclaredAnnotation(TestCaseIdKey.class);
```

Comment:
move it to a separate method with a clear name

Reply:
done